modules/listener/listener.tf

```diff
@@ -0,0 +1,191 @@
+  statement_id  = "AllowExecutionFromEventBridge"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.listener.function_name
+  principal     = "events.amazonaws.com"
+  source_arn    = aws_cloudwatch_event_rule.logs.arn
+}
+
+resource "aws_iam_role" "listener" {
+  name = format("%s-listener-role", var.prefix)
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          Service = "lambda.amazonaws.com"
+        }
+        Action = "sts:AssumeRole"
+      },
+    ]
+  })
+
+  managed_policy_arns = [
+    aws_iam_policy.listener.arn
+  ]
+
+  tags = {
+    PartOf    = var.prefix
+    Platform  = "Axiom"
+    Component = "axiom-cloudwatch-listener"
+  }
+}
+
+resource "aws_iam_policy" "listener" {
+  name   = format("%s-listener-lambda-policy", var.prefix)
+  path   = "/"
+  policy = data.aws_iam_policy_document.listener.json
+  tags = {
+    PartOf    = var.prefix
+    Platform  = "Axiom"
+    Component = "axiom-cloudwatch-listener"
+  }
+}
+
+resource "aws_cloudwatch_log_group" "listener" {
+  name              = format("/aws/axiom/%s-listener", var.prefix)
+  retention_in_days = 1
+  tags = {
+    PartOf    = var.prefix
+    Platform  = "Axiom"
+    Component = "axiom-cloudwatch-listener"
+  }
+}
+
+data "aws_iam_policy_document" "cloudtrail" {
+  count = var.enable_cloudtrail ? 1 : 0
+  statement {
+    sid    = "AWSCloudTrailAclCheck20150319"
+    effect = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+    actions   = ["s3:GetBucketAcl"]
+    resources = [aws_s3_bucket.cloudtrail[0].arn]
+  }
+  statement {
+    sid    = "AWSCloudTrailWrite20150319"
+    effect = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+    actions = ["s3:PutObject"]
+    resources = [
+      format("%s/AWSLogs/%s/*", aws_s3_bucket.cloudtrail[0].arn, data.aws_caller_identity.current.account_id)
+    ]
+    condition {
+      test     = "StringEquals"
+      variable = "s3:x-amz-acl"
+      values   = ["bucket-owner-full-control"]
+    }
+  }
+}
+
+resource "aws_s3_bucket_policy" "cloudtrail" {
+  count      = var.enable_cloudtrail ? 1 : 0
+  depends_on = [aws_s3_bucket.cloudtrail]
+  bucket     = aws_s3_bucket.cloudtrail[0].id
+  policy     = data.aws_iam_policy_document.cloudtrail[0].json
+}
+
+
+resource "aws_s3_bucket" "cloudtrail" {
+  count = var.enable_cloudtrail ? 1 : 0
+  # aws_s3_bucket_acl    = "BucketOwnerFullControl"
+  bucket = format("%s-cloudtrail", var.prefix)
+
+  tags = {
+    PartOf    = var.prefix
+    Platform  = "Axiom"
+    Component = "axiom-cloudwatch-listener"
+  }
+}
+
+
+// TODO:
+// - Cloudformation has IsLogging: true, but no idea what does it do
+// - Maybe we can make use event_selector to use cost?
+resource "aws_cloudtrail" "cloudtrail" {
```

Comment:
I think this has to be removed from here and from Cloudformation. We should not enable a CloudTrail Trail. CloudTrail itself and cloudtrail events are always enabled as far as I know. They don't require a Trail to be enabled, and if they do, its something users should/would manage globally, not as part of a module like this IMO.

Reply:
> CloudTrail itself and cloudtrail events are always enabled as far as I know.

So Users don't have to enable this at all? Should we mention something about it in the docs?

Reply:
I don't think they have to enable it at all. I only doubt because of what you mentioned during the test, but I would like to replicate because I don't think a CloudTrail Trail is required for CloudTrail based events to work.

Reply:
I have tested again on our dev account without creating a new trail and it worked. Can we test somewhere on a new account or something where there is no trails at all?

Reply:
I don't think we can disable it on a particular account, as its enabled for the entire org. @a-khaledf do we have any test accounts outside the org?

Reply:
I removed the *Cloudtrail* resources from terraform module and Cloudformation. If you are confident with it we can move on.